### ISSUE TITLE
Activation results in blank page

### ISSUE THREAD

**Issue by DylanCodeCabin** [CodeCabin/wp-google-maps#677]
Find transcript below:

---

Hello,

When the WP Google Maps plugin is packed in a WP theme, along with other plugins and managed through the TGM Plugin, once the plugin is activated through the required plugins screen of TGM, it results in a blank screen. 

I have tracked down the source of the issue here:
https://plugins.trac.wordpress.org/browser/wp-google-maps/trunk/wpGoogleMaps.php#L744

The lines which clear the output header and redirect the user, are breaking the TGM activation process and result in a blank screen. This means that the user will be left with a blank screen and won't know what happened or what to do. Were the plugins activated? Was there some kind of error?:
```
// clean the output header and redirect the user

@ob_flush();

@ob_end_flush();

@ob_end_clean();

		            

exit( wp_redirect( admin_url( 'admin.php?page=wp-google-maps-menu&action=welcome_page' ) ) );
```

I suggest a conditional is placed in the wpgmza_redirect_on_activate() function to check whether the plugin is being activated through the default WP plugins page or via the TGM plugins activation page and disable the redirect.

Something along the lines of:
```
function wpgmza_redirect_on_activate( $plugin ) {

    if( $plugin == plugin_basename( __FILE__ ) ) {

        if ( !get_option( "WPGM_V6_FIRST_TIME" ) ) {



            $current_screen = get_current_screen();

						// If we are in the TGM plugins activation screen, don't redirect:

            if ( $current_screen && $current_screen->id != "appearance_page_install-required-plugins" ){

              update_option( "WPGM_V6_FIRST_TIME", true );

              // clean the output header and redirect the user

              @ob_flush();

              @ob_end_flush();

              @ob_end_clean();

              exit( wp_redirect( admin_url( 'admin.php?page=wp-google-maps-menu&action=welcome_page' ) ) );

            }



        }

    }

}
```

Maybe a more sophisticated method exists (through hooks, etc.) but this issue has been bugging our customers for months as it creates a conflict with the TGM plugin .

Thank you

**Comment by NickDuncanZA** [comment]
High priority please

**Comment by PerryRylance** [comment]
Suggested fix implemented in 8.0.12
